### ISSUE TITLE
hset command does not give proper return values

### ISSUE THREAD

**Issue by c-simmons** [brigade/mock_redis#136]
The command hset from redis-rb returns a different value than mock_redis. Redis returns a 1 when the key is new in the hash, or 0 if the key exists and was changed. The redis client gem converts these return values to true and false respectively, mock_redis universally returns true.

Below is an irb session that demonstrates the issue.
```
irb(main):003:0> r = Redis.new
irb(main):004:0> mr = MockRedis.new
irb(main):005:0> r.hset('myhash', 'key', 'first')
=> true
irb(main):006:0> mr.hset('myhash', 'key', 'first')
=> true
irb(main):007:0> r.hset('myhash', 'key', 'second')
=> false
irb(main):008:0> mr.hset('myhash', 'key', 'second')
=> true
```

**Comment by sds** [comment]
Hey @c-simmons, thanks for the report!

Would love to merge a pull request fixing this!

**Comment by polpak** [comment]
I'll see about submitting one.

**Comment by polpak** [comment]
Ugh.. Your readme says the gem is tested against Redis 3, but Travis looks like it's testing against Redis 4. I'm going to assume we want to do the latter, so I'll update the PR to fix the other failing tests and update the Readme as well.

**Comment by sds** [comment]
Fixed by #139. Thanks @polpak!
